### DISCUSSION TITLE
How to re-implement the C/C++ functions for MCU rather than using RUST.

### DISCUSSION THREAD

**Discussion by YTomTJ** [slint-ui/slint#2756] [Q&A] [answered]
I'm not familiar with RUST on developping for MCU, but I want to make Slint work on my C/C++ MCU project. My idea is rendering the UI into a memory and update it to the device (no touching avaliable) via bus in certain refreshing frequency. So problem is how to re-implement the C/C++ interface functions like slint_cpp.dll did ?

**Comment by ogoffart** [comment, marked as answer]
Our C++ API for MCU is not yet finished. See issue https://github.com/slint-ui/slint/issues/2136
I wouldn't really recommand trying to stream the UI, although this can be possible with experimental API.
What MCU are you targeting? What exactly are you trying to do?

**Reply by YTomTJ** [reply]
I'm using STM32F411 at present. I just find Slint so powerful and want to make a trial on MCU. Maybe such trial should be useful for my future work. Anyway, I'll try some other ways. Thanks for your reply.
